pkg/controller/projects/members/controller.go

```diff
@@ -187,3 +211,13 @@ func isoTimeToString(i interface{}) string {
 	}
 	return ""
 }
+
+func lateInitializeProjectMember(local *v1alpha1.MemberParameters, external *gitlab.ProjectMember) {
+	if external == nil {
+		return
+	}
+
+	if local.UserID == nil {
+		local.UserID = &external.ID
```

Comment:
This won't work because `local.UserID` is always updated to the userID found by the given userName.

Reply:
deleted lateInitializeProjectMember func